internal/api/v20240610preview/external_auth_methods.go

```diff
@@ -79,21 +79,32 @@ func (h *ExternalAuth) Normalize(out *api.HCPOpenShiftClusterExternalAuth) {
 	}
 }
 
+func (c *ExternalAuth) GetVisibility(path string) (api.VisibilityFlags, bool) {
+	flags, ok := externalAuthVisibilityMap[path]
+	return flags, ok
+}
+
+func (c *ExternalAuth) ValidateVisibility(current api.VersionedCreatableResource[api.HCPOpenShiftClusterExternalAuth], updating bool) []arm.CloudErrorBody {
+	var structTagMap = api.GetStructTagMap[api.HCPOpenShiftClusterExternalAuth]()
+	return api.ValidateVisibility(c, current.(*ExternalAuth), externalAuthVisibilityMap, structTagMap, updating)
+}
+
 func (c *ExternalAuth) ValidateStatic(current api.VersionedHCPOpenShiftClusterExternalAuth, updating bool, request *http.Request) *arm.CloudError {
-	var normalized api.HCPOpenShiftClusterExternalAuth
 	var errorDetails []arm.CloudErrorBody
 
-	// Pass the embedded ExternalAuth struct so the
-	// struct field names match the externalAuthStructTagMap keys.
-	errorDetails = api.ValidateVisibility(
-		c.ExternalAuth,
-		current.(*ExternalAuth).ExternalAuth,
-		externalAuthStructTagMap, updating)
+	errorDetails = c.ValidateVisibility(current, updating)
+
+	// Proceed with additional validation only if visibility validation has
+	// passed. This avoids running further checks on changes we already know
+	// to be invalid and prevents the response body from becoming overwhelming.
+	if len(errorDetails) == 0 {
+		var normalized api.HCPOpenShiftClusterExternalAuth
```

Comment:
No blocking but just my 2 cents: I would do the opposite here and give a user all the information rather than "whack-a-mole". Maybe a question for product.

Reply:
I definitely want to avoid "whack-a-mole".

My thinking was just to partition validation into three progressive phases.

1. Are you even allowed to set/change these fields?
2. Are the values you're setting these fields to valid (syntactically, within acceptable limits, etc.)?
3. Do these changes violate invariants across the whole resource (or its parent)?

Each phase can still report multiple errors, but your changes have to pass one phase to get to the next.

But yes, it's debatable if this is better than everything at once.